### PR TITLE
Update UCX-Py version to 0.21

### PR DESCRIPTION
**Pull request by pentschev**

### CODE REVIEW

**Comment by pentschev** [comment]
Waiting for https://github.com/rapidsai/cugraph/pull/1649 to be merged first.

**Review by rlratzel — APPROVED** [review]
Thanks!

File: ci/release/update-version.sh [hunk 1]
```diff
@@ -50,6 +50,6 @@ for FILE in conda/environments/*.yml; do
    sed_runner "s/rmm=${CURRENT_SHORT_TAG}/rmm=${NEXT_SHORT_TAG}/g" ${FILE};
    sed_runner "s/dask-cuda=${CURRENT_SHORT_TAG}/dask-cuda=${NEXT_SHORT_TAG}/g" ${FILE};
    sed_runner "s/dask-cudf=${CURRENT_SHORT_TAG}/dask-cudf=${NEXT_SHORT_TAG}/g" ${FILE};
-   sed_runner "s/ucx-py=${CURRENT_SHORT_TAG}/ucx-py=${NEXT_SHORT_TAG}/g" ${FILE};
+   sed_runner "s/ucx-py=0.21/ucx-py=0.21/g" ${FILE};
```
Comment (ajschmidt8) [inline ci/release/update-version.sh:53]: this is a no-op, so we can remove this line.

**Review by ajschmidt8 — CHANGES_REQUESTED** [review]

**Review by BradReesWork — APPROVED** [review]

**Review by ajschmidt8 — APPROVED** [review]

**Comment by ajschmidt8** [comment]
@gpucibot merge

**Comment by pentschev** [comment]
Seems like there are several potentially legit cuGraph failures here, @rlratzel @BradReesWork could you take a look?

**Comment by pentschev** [comment]
Oh, actually those seem to be because we're installing Dask/Distributed 2021.5.1, but we need 2021.6.0 or main for branch-21.08. It seems that this was covered by https://github.com/rapidsai/cugraph/blob/6aab21f01173b0b430197d15bdc700331ebe70c3/ci/gpu/build.sh#L75-L77 but something is downgrading them after that, perhaps we need https://github.com/rapidsai/dask-cuda/pull/645 first.

**Comment by pentschev** [comment]
Actually no, https://github.com/rapidsai/dask-cuda/pull/645 won't help because it targets branch-21.06. I think something in the following block is downgrading Dask/Distributed: 
https://github.com/rapidsai/cugraph/blob/6aab21f01173b0b430197d15bdc700331ebe70c3/ci/gpu/build.sh#L89-L113

I'm not sure what package is the cause for downgrade, but perhaps we could move the Dask install after cuGraph? Also, this will need to wait for https://github.com/rapidsai/dask-cuda/pull/647, as https://github.com/rapidsai/dask-cuda/pull/645 just got merged and that pins 2021.5.1 in Dask-CUDA.

**Comment by pentschev** [comment]
rerun tests

**Comment by pentschev** [comment]
Rerunning tests now that cuDF and integration repo have been updated.

**Comment by ajschmidt8** [comment]
looks like this might be blocked by #1637 now. that one should hopefully be merged soon.

**Comment by ajschmidt8** [comment]
@pentschev, can you pull the latest changes here now that #1637 is merged?

**Comment by pentschev** [comment]
> can you pull the latest changes here now that #1637 is merged?

Done.

**Pull request merged**